Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.Data.cs

```diff
@@ -33,6 +41,55 @@ public UvMeasure(Orientation orientation, double width, double height)
                     V = width;
                 }
             }
+
+            public UvMeasure Add(double u, double v)
+                => new UvMeasure { U = U + u, V = V + v };
+        }
+
+        private struct UvRect
+        {
+            public UvMeasure Position { get; set; }
+
+            public UvMeasure Size { get; set; }
+
+            public Rect ToRect(Orientation orientation)
+            {
+                switch (orientation)
+                {
+                    case Orientation.Vertical:
+                        return new Rect(Position.V, Position.U, Size.V, Size.U);
+                    case Orientation.Horizontal:
+                        return new Rect(Position.U, Position.V, Size.U, Size.V);
+                    default:
+                        throw new NotSupportedException();
+                }
+            }
+        }
+
+        private struct Row
+        {
+            private readonly List<UvRect> _childrenRects;
+            private UvMeasure _rowSize;
+
+            public Row(List<UvRect> childrenRects, UvMeasure size)
+            {
+                _childrenRects = childrenRects;
+                _rowSize = size;
+            }
+
+            public IReadOnlyList<UvRect> ChildrenRects => _childrenRects;
```

Comment:
Doing this will prevent the C# compiler from using the fast `List<T>.Enumerator` type when going through children in the actual control, and will cause the enumerator to be boxed and to emit a bunch of virtual calls. Given that this type is internal, I'd suggest just removing this and exposing the `List<UvRect>` directly to allow for faster iteration.